model-optimizer/mo/moc_frontend/extractor.py

```diff
@@ -1,35 +1,70 @@
 # Copyright (C) 2018-2021 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-import logging as log
 import re
-from collections import defaultdict
-from copy import copy
-
-import numpy as np
 
+from mo.front.extractor import raise_no_node, raise_node_name_collision
 from mo.utils.error import Error
 
-from ngraph.frontend import InputModel  # pylint: disable=no-name-in-module,import-error
+from ngraph.frontend import InputModel, Place  # pylint: disable=no-name-in-module,import-error
+
+import numpy as np
 
 
 def decode_name_with_port(input_model: InputModel, node_name: str):
     """
     Decode name with optional port specification w/o traversing all the nodes in the graph
-    TODO: in future node_name can specify input/output port groups and indices (58562)
+    TODO: in future node_name can specify input/output port groups as well as indices (58562)
     :param input_model: Input Model
     :param node_name: user provided node name
     :return: decoded place in the graph
     """
-    # Check exact match with one of the names in the graph first
+    found_nodes = []
+    found_node_names = []
+
     node = input_model.get_place_by_tensor_name(node_name)
-    if node:
-        return node
+    found_node_names.append('Tensor:' + node_name) if node else None
+    found_nodes.append(node) if node else None
```

Comment:
nit, but I would recommend to refactor the code the following way to increase readability. Also, the current approach is not commonly used in the MO so I recommend to change the code to follow the existing MO code style.

```python
if node:
    found_node_names.append('Tensor:' + node_name)
    found_nodes.append(node)
```

Reply:
Fixed